src/OpenLoco/src/Windows/CompanyFaceSelection.cpp

```diff
@@ -135,43 +145,37 @@ namespace OpenLoco::Ui::Windows::CompanyFaceSelection
     // 0x4352BB
     static void getScrollSize([[maybe_unused]] Window& self, [[maybe_unused]] const uint32_t scrollIndex, [[maybe_unused]] uint16_t* const scrollWidth, uint16_t* const scrollHeight)
     {
-        *scrollHeight = _numberCompetitorObjects * kRowHeight;
+        *scrollHeight = static_cast<uint16_t>(_competitorList.size()) * kRowHeight;
     }
 
     static bool isInUseCompetitor(const uint32_t objIndex)
     {
         return std::find(_inUseCompetitors.begin(), _inUseCompetitors.end(), objIndex) != _inUseCompetitors.end();
     }
 
-    struct ObjectRow
-    {
-        ObjectManager::ObjectIndexEntry object;
-        int16_t rowIndex;
-    };
-    static ObjectRow getObjectFromSelection(const int16_t& y)
+    static ObjectManager::ObjIndexPair getObjectFromSelection(const int16_t& y)
     {
         const int16_t rowIndex = y / kRowHeight;
-        const auto objects = ObjectManager::getAvailableObjects(ObjectType::competitor);
 
-        if (rowIndex < 0 || static_cast<uint16_t>(rowIndex) >= objects.size())
+        if (rowIndex < 0 || static_cast<uint16_t>(rowIndex) >= _competitorList.size())
         {
-            return { ObjectManager::ObjectIndexEntry{}, -1 };
+            return { -1, ObjectManager::ObjectIndexEntry{} };
```

Comment:
Do we not have a constant for -1? If not we should introduce kInvalidObjectIndex